doc/releases/release-notes-4.3.rst

```diff
@@ -48,6 +48,10 @@ https://docs.zephyrproject.org/latest/security/vulnerabilities.html
 API Changes
 ***********
 
+* Kernel
+
+  * :c:func:`device_init`
+
```

Comment:
Hey, this block is supposed to be empty, As in this should only be the top title for "Removed APIs and options" etc.. API Changes are obviously migration-guide material, no point duplicating it here, I'll followup dropping the rtio one sorry about it but can you please drop this for now?